### PR TITLE
Add code coverage reporting and improve CI

### LINKED ISSUE #50: Add code coverage tracking through CI
**Is your feature request related to a problem? Please describe.**
We don't currently have a way of tracking code coverage.

**Describe the solution you'd like**
I'd like CI to tell us the code coverage of the main branch and of PRs.

### PR DESCRIPTION
**Pull request by vincentmacri**
This fixes #50.

It also cleans up the naming of the different CI workflows.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
   build:
-    runs-on: ubuntu-16.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-latest, windows-latest, macos-latest]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v1
```
Comment (tyxchen) [inline .github/workflows/build.yml:16]: Curious why the downgrade here?

Comment (vincentmacri) [reply .github/workflows/build.yml]: Don't know, that was an accident.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Install dependencies
+      if: steps.npm-cache.outputs.cache-hit != 'true'
+      run: npm install --no-audit --no-fund
+    
+    - name: Check type errors
+      run: npx tsc --noEmit
+
+    - name: Run tests
+      if: matrix.os != 'ubuntu-16.04'
+      run: npm test -- --coverage
```
Comment (tyxchen) [inline .github/workflows/test.yml:42]: Do we need to generate a coverage report for non-16.04 pipelines?

Comment (vincentmacri) [reply .github/workflows/test.yml:42]: No, it seems to just upload the same thing.

I think I could easily get it to do code coverage for the other operating systems, but doing a coverage test is slower, and I only care about non-16.04 for making sure devs don't have trouble.

In theory there could be something that only happens in certain operating systems. I could try doing coverage for all if you want.

Comment (vincentmacri) [reply .github/workflows/test.yml:42]: Oh I see what you meant now. That's a typo. Will fix.

**Pull request merged**